Tight react peer dependency for `@dxos/react-ui`?

Trying to add `@dxos/react-client` to a react `v18.3.1` app and getting this:


npm install @dxos/react-client
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: savesafe@0.0.0
npm error Found: react@18.3.1
npm error node_modules/react
npm error   react@"^18.3.1" from the root project
npm error   peer react@">= 16.8" from @phosphor-icons/react@2.1.7
npm error   node_modules/@phosphor-icons/react
npm error     peer @phosphor-icons/react@"^2.1.5" from @dxos/react-ui@0.7.0
npm error     node_modules/@dxos/react-ui
npm error       peer @dxos/react-ui@"0.7.0" from @dxos/react-client@0.7.0
npm error       node_modules/@dxos/react-client
npm error         @dxos/react-client@"*" from the root project
npm error   1 more (react-dom)
npm error
npm error Could not resolve dependency:
**npm error peer react@"~18.2.0" from @dxos/react-ui@0.7.0**
npm error node_modules/@dxos/react-ui
npm error   peer @dxos/react-ui@"0.7.0" from @dxos/react-client@0.7.0
npm error   node_modules/@dxos/react-client
npm error     @dxos/react-client@"*" from the root project

Maybe the react peer dep could be setup as 18.x?

--- a/packages/sdk/react-client/package.json
+++ b/packages/sdk/react-client/package.json
@@ -104,13 +104,13 @@
     "react-dom": "~18.2.0"
   },
   "peerDependencies": {
-    "@dxos/react-ui": "workspace:*",
-    "@dxos/react-ui-theme": "workspace:*",
     "react": "~18.2.0",
     "react-dom": "~18.2.0"
   },
   "optionalDependencies": {
-    "@dxos/random": "workspace:*"
+    "@dxos/random": "workspace:*",
+    "@dxos/react-ui": "workspace:*",
+    "@dxos/react-ui-theme": "workspace:*"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/sdk/shell/package.json
+++ b/packages/sdk/shell/package.json
@@ -86,11 +86,13 @@
   },
   "peerDependencies": {
     "@dxos/react-client": "workspace:*",
-    "@dxos/react-ui": "workspace:*",
-    "@dxos/react-ui-theme": "workspace:*",
-    "@phosphor-icons/react": "^2.1.5",
     "react": "~18.2.0",
     "react-dom": "~18.2.0"
+  },
+  "optionalDependencies": {
+    "@dxos/react-ui": "workspace:*",
+    "@dxos/react-ui-theme": "workspace:*",
+    "@phosphor-icons/react": "^2.1.5"
   },
   "publishConfig": {
     "access": "public"

